presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -1522,6 +1523,11 @@ public SystemSessionProperties(
                         "Track history based plan statistics service in query optimizer",
                         featuresConfig.isTrackHistoryBasedPlanStatistics(),
                         false),
+                booleanProperty(
+                        TRACK_COMPLETE_STAGES_STATS_FROM_FAILED_QUERY,
```

Comment:
not sure if we need "COMPLETE_STAGES" as part of the property name - perhaps leave it out and only specify it in the documentation string
another idea would be to turn this property around and make it track_history_from_successful_queries_only

Reply:
> not sure if we need "COMPLETE_STAGES" as part of the property name - perhaps leave it out and only specify it in the documentation string another idea would be to turn this property around and make it track_history_from_successful_queries_only

So session params should be default true to reflect the preferred default behavior for clarity. I prefer true values for session params as default. So I say stick with failed_queries version

Reply:
I try to avoid words with negative meaning in the flags because it makes it hard to reason about enabling/disabling but it makes sense to leave it as in this case. We would probably only need to disable it if there is a problem with it